### PR TITLE
Controllers refactor

### PR DESCRIPTION
**Pull request by rborcat**
Controllers refactor and added an exception validator

### CODE REVIEW

File: src/main/java/com/upversionlab/controller/CompanyController.java [hunk 1]
```diff
@@ -14,15 +15,16 @@
 @RequestMapping("/companies")
 public class CompanyController {
     private Map<Integer, Company> companies = new HashMap<>();
+    private static final String COMPANY_RESOURCE_NAME = Company.class.getSimpleName();
```
Comment (vruzeda) [inline src/main/java/com/upversionlab/controller/CompanyController.java]: Constants should come first (and be probably be separated by an empty line from other attributes).

File: src/main/java/com/upversionlab/controller/IngredientController.java [hunk 2]
```diff
@@ -11,17 +12,19 @@
  * Created by rborcat on 1/3/2017.
  */
 @RestController
+@RequestMapping("/ingredients")
 public class IngredientController {
     private Map<Integer, Ingredient> ingredients = new HashMap<>();
+    private static final String INGREDIENT_RESOURCE_NAME = Ingredient.class.getSimpleName();
```
Comment (vruzeda) [inline src/main/java/com/upversionlab/controller/IngredientController.java]: Constants should come first (and be probably be separated by an empty line from other attributes).

File: src/main/java/com/upversionlab/controller/ProductController.java [hunk 3]
```diff
@@ -14,15 +15,16 @@
 @RequestMapping("/products")
 public class ProductController {
     private Map<Integer, Product> products = new HashMap<>();
+    private static final String PRODUCT_RESOURCE_NAME = Product.class.getSimpleName();
```
Comment (vruzeda) [inline src/main/java/com/upversionlab/controller/ProductController.java]: Constants should come first (and be probably be separated by an empty line from other attributes).

**Review by vruzeda — CHANGES_REQUESTED** [review]

File: src/main/java/com/upversionlab/controller/ProductController.java (hunk 3 above)
Comment (rborcat) [reply src/main/java/com/upversionlab/controller/ProductController.java]: ok

File: src/main/java/com/upversionlab/controller/IngredientController.java (hunk 2 above)
Comment (rborcat) [reply src/main/java/com/upversionlab/controller/IngredientController.java]: ok

File: src/main/java/com/upversionlab/controller/CompanyController.java (hunk 1 above)
Comment (rborcat) [reply src/main/java/com/upversionlab/controller/CompanyController.java]: ok

**Review by vruzeda — APPROVED** [review]

**Pull request merged**